yt/visualization/profile_plotter.py

```diff
@@ -287,39 +269,30 @@ def save(self, name=None, suffix=None, mpl_kwargs=None):
             iters = zip(range(len(unique)), sorted(unique))
         else:
             iters = self.plots.items()
-        if not suffix:
-            suffix = "png"
-        suffix = f".{suffix}"
-        fullname = False
+
         if name is None:
             if len(self.profiles) == 1:
-                prefix = self.profiles[0].ds
-            else:
-                prefix = "Multi-data"
-            name = f"{prefix}{suffix}"
-        else:
-            sfx = get_image_suffix(name)
-            if sfx != "":
-                suffix = sfx
-                prefix = name[: name.rfind(suffix)]
-                fullname = True
+                name = str(self.profiles[0].ds)
             else:
-                prefix = name
+                name = "Multi-data"
+
+        name = validate_image_name(name, suffix)
+        prefix, suffix = os.path.splitext(name)
+
         xfn = self.profiles[0].x_field
         if isinstance(xfn, tuple):
             xfn = xfn[1]
-        fns = []
+
+        names = []
         for uid, plot in iters:
             if isinstance(uid, tuple):
                 uid = uid[1]
-            if fullname:
```

Comment:
I note that I broke that part of the logic tree. I think this is one of the places where we implement a "smart" pattern to allow a full path name (that is, the exact desired name of the output file, with an extension included) be passed, while this is not the general way output file names are constructed.
Meanwhile, it doesn't make much tore use a unique exact name inside a loop as is done here.
All in all, I think the code would be much more maintainable if we renounced inferring an extension from the `name` argument, _at least_ when more than one image file is to be saved.